app/modules/booking/helpers.ts

```diff
@@ -0,0 +1,29 @@
+import { isAfter, isSameDay } from "date-fns";
+
+/**
+ * This function checks if the booking is being early checkout
+ */
+export function isBookingEarlyCheckout(from: string | Date): boolean {
```

Comment:
This only checks date. Should also check time and even if its 5 minutes earlier it should adjust it.